components/style/properties/declaration_block.rs

```diff
@@ -169,19 +177,33 @@ impl PropertyDeclarationBlock {
     }
 
     /// Adds or overrides the declaration for a given property in this block,
-    /// without taking into account any kind of priority. Returns whether the
-    /// declaration block is actually changed.
+    /// except if an existing declaration for the same property is more important.
+    pub fn push(&mut self, declaration: PropertyDeclaration, importance: Importance) {
+        self.push_common(declaration, importance, false);
+    }
+
+    /// Adds or overrides the declaration for a given property in this block,
+    /// Returns whether the declaration block is actually changed.
     pub fn set_parsed_declaration(&mut self,
                                   declaration: PropertyDeclaration,
                                   importance: Importance) -> bool {
+        self.push_common(declaration, importance, true)
+    }
+
+    fn push_common(&mut self, declaration: PropertyDeclaration, importance: Importance,
```

Comment:
I feel like we could have more comments about this new design here, mentioning the word "deduplicate" at least once 😄 (for greppability)